### ISSUE TITLE
null leads to an invalid validation result

### ISSUE THREAD

**Issue by tjamin** [ayosdev/ng-iban-custom#8]
Nice directive. Thanks a lot for sharing!

I found a tiny problem with it. If the value of the bound model is null, the validation returns undefined instead of a boolean and therefore $valid/$invalid of the FormController becomes undefined also which breaks angular form validation.

[Here](https://github.com/ayosdev/ng-iban-custom/blob/master/dist/ng-iban.js#L314) we need a case for `viewValue == null` to set the variable `valid` to true. And why true and not false? If a field is a required field that required-validation has to be handled by ng-required.

**Comment by Ugoku** [comment]
I think the same goes if the value is `''` (empty string) and `required=false`. In that case, `viewValue != null` but `''` is not a valid IBAN so the validation fails.

If I'm misunderstanding and this should be a separate issue, I apologise and will file a separate issue.

**Comment by tjamin** [comment]
You are right! A few hours after I posted the issue we recognized the same as you do. We changed the code locally because I think this repository is dead. We extended our solution with a new  attribute: `ng-iban-validate-when`. The iban will only be validated if that attribute evaluates to true:

**Code**

``` javascript
    ctrl.$validators.ngIban = function (viewValue) {
        var parsed, valid;
        if (viewValue == null || viewValue === '' || (attrs.ngIbanValidateWhen && !scope.$eval(attrs.ngIbanValidateWhen))) {
            valid = true;
        } else {
            valid = isValidIban(viewValue);
            if (valid) {
                parsed = parseIban(viewValue);
                if (parsed !== viewValue) {
                    ctrl.$setViewValue(parsed);
                    ctrl.$render();
                }
            }
        }
        return valid;
    };
```

**Usage**

``` html
<input type="text"
       name="iban"
       ng-model="m.iban"
       ng-required="m.selectedPaymentMethod.id == 'debit'"
       ng-iban="DE"
       ng-iban-validate-when="m.selectedPaymentMethod.id == 'debit'" />
```

**Comment by brauliodiez** [comment]
I think the best could be to fork the repo add a test showing the issue, the placing the solutions.

As far as I know there is a team involved on this project (some months ago the accepted a PR from my side).

**Comment by Celestz** [comment]
PRs are very welcome :)
